### PR TITLE
Update Qt to 5.15.7

### PR DESCRIPTION
**Pull request by chromatic**
This may be a great or a terrible idea. Let's see where it goes.

### CODE REVIEW

**Comment by chromatic** [comment]
CI runs seem to be failing because of Node 12 deprecations. I thought we'd fixed that with PR #3140 .

**Comment by alamshafil** [comment]
Looks like this link: `https://download.qt.io/new_archive/qt/5.15/5.15.7/submodules/qtbase-everywhere-opensource-src-5.15.7.tar.xz` does not exist.

Looks like it should be using `https://download.qt.io/official_releases/qt/5.15/5.15.7/submodules/qtbase-everywhere-opensource-src-5.15.7.tar.xz`

File: depends/packages/qt.mk [hunk 1]
```diff
@@ -1,21 +1,21 @@
 PACKAGE=qt
-$(package)_version=5.7.1
-$(package)_download_path=http://download.qt.io/new_archive/qt/5.7/$($(package)_version)/submodules
```
Comment (alamshafil) [inline depends/packages/qt.mk:3]: Will this fix CI?
`$(package)_download_path=https://download.qt.io/official_releases/qt/5.15/$($(package)_version)/submodules`

Comment (chromatic) [reply depends/packages/qt.mk:3]: I was *certain* I changed that. Good catch. Let's try it.

Comment (chromatic) [reply depends/packages/qt.mk:3]: Ah, I didn't run `make -j9 -C depends` locally, which will give me more information. I'll get back to this in a day or two.

Comment (chromatic) [reply depends/packages/qt.mk:3]: I think, in addition to the download paths changing, the format of the downloadable SHA-256 checksums has changed too:

https://download.qt.io/official_releases/qt/5.15/5.15.7/submodules/qtcharts-everywhere-opensource-src-5.15.7.tar.xz.sha256

I'm lost in a twisty maze of autoconf definitions right now though.

File: depends/packages/qt.mk [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-$(package)_suffix=opensource-src-$($(package)_version).tar.gz
+$(package)_version=5.15.7
+$(package)_download_path=http://download.qt.io/official_releases/qt/5.15/$($(package)_version)/submodules
+$(package)_suffix=everywhere-opensource-src-$($(package)_version).tar.xz
 $(package)_file_name=qtbase-$($(package)_suffix)
-$(package)_sha256_hash=95f83e532d23b3ddbde7973f380ecae1bac13230340557276f75f2e37984e410
+$(package)_sha256_hash=a05edd00b2a1ba99c85b3fe876fe1c23d63f0a9bbca7df52bc47393cfd8c809c7
```
Comment (alamshafil) [inline depends/packages/qt.mk:6]: Looks like there is a extra `a` here.

My output:
```bash
sha256sum qtbase-everywhere-opensource-src-5.15.7.tar.xz 
05edd00b2a1ba99c85b3fe876fe1c23d63f0a9bbca7df52bc47393cfd8c809c7  qtbase-everywhere-opensource-src-5.15.7.tar.xz
```

Comment (alamshafil) [reply depends/packages/qt.mk:6]: I checked the sha256sum of the other archives and they look fine.

**Comment by patricklodder** [comment]
Closing this for now, like #2339 and #2365

**Pull request closed without merging**